src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBarracks.java

```diff
@@ -0,0 +1,198 @@
+package com.minecolonies.coremod.colony.buildings;
+
+import com.minecolonies.blockout.views.Window;
+import com.minecolonies.coremod.blocks.BlockHutBarracksTower;
+import com.minecolonies.coremod.blocks.ModBlocks;
+import com.minecolonies.coremod.client.gui.WindowBarracksBuilding;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyView;
+import com.minecolonies.coremod.tileentities.TileEntityColonyBuilding;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.Tuple;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+
+/**
+ * Building for the Barracks.
+ */
+public class BuildingBarracks extends AbstractBuilding
+{
+    /**
+     * General Barracks description key.
+     */
+    private static final String BARRACKS = "Barracks";
+
+    /**
+     * Max hut level of the Barracks.
+     */
+    private static final int BARRACKS_HUT_MAX_LEVEL = 5;
+
+    /**
+     * Tower position offset.
+     */
+    private static final int TOWER_OFFSET = 8;
+
+    /**
+     * Constructor for the Barracks building.
+     *
+     * @param c Colony the building is in.
+     * @param l Location of the building.
+     */
+    public BuildingBarracks(final Colony c, final BlockPos l)
+    {
+        super(c, l);
+    }
+
+    @Override
+    public void onUpgradeComplete(final int newLevel)
+    {
+        final World world = getColony().getWorld();
+        if (world != null && newLevel < 5)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)